packages/cli/generators/model/index.js

```diff
@@ -297,6 +298,52 @@ module.exports = class ModelGenerator extends ArtifactGenerator {
         if (this.artifactInfo.allowAdditionalProperties) {
           Object.assign(this.artifactInfo.modelSettings, {strict: false});
         }
+      })
+      .catch(err => {
+        debug(`Error during model strict mode prompt: ${err}`);
+        return this.exit(err);
+      });
+  }
+
+  async promptRelations() {
+    if (this.shouldExit()) return false;
+    const models = [];
+
+    // get the list of current models
+    try {
+      debug(`model list dir ${this.artifactInfo.modelDir}`);
+      const modelList = await utils.getArtifactList(
+        this.artifactInfo.modelDir,
+        'model',
+      );
+      debug(`modelList ${modelList}`);
+
+      if (modelList && modelList.length > 0) {
+        models.push(...modelList);
+        debug(`models: ${models}`);
+      }
+    } catch (err) {
+      debug(`error ${err}`);
+      return this.exit(err);
+    }
+
+    // skip if there are no other models
+    if (!models.length) return;
+
+    return this.prompt([
+      {
+        name: 'relations',
+        message: 'Is this model part of any relations?',
```

Comment:
Very recently, we landed `lb4 relation` command - see https://github.com/strongloop/loopback-next/commit/75939a4a144b3068ff2890394c178faad58d7458.

I find it weird to have `lb4 model` prompting for model relations, when there is a dedicated command that does better job at scaffolding relation-based code.

Is this part addressing the following acceptance criteria?

> Update lb4 model to start emitting `{ModelName}Relations` interfaces

As I was envisioning the solution, an empty interface should be emitted for all newly-created models, so that we can _Update `lb4 repository` to include `{ModelName}Relations` in the generic parameters of `DefaultCrudRepository` we are inheriting from_

Entries for individual relations would be added to `{ModelName}Relations` either manually by the application developer or by `lb4 repository` command.

My conclusions:
- The answer to your _Question: should all models include a relations interface?_ is YES.
- These new `lb4 model` prompts should be removed.
- Ideally, `lb4 repository` should be improved to add entries to relations interfaces. This was not part of the original acceptance criteria and I don't know how much effort would be required, it may be best to leave this part out of scope of this pull request and possibly out of scope of #2960 too.

What do you think?

Reply:
@bajtos thanks for the comment!

> Is this part addressing the following acceptance criteria?
> 
> > Update `lb4 model` to start emitting `{ModelName}Relations` interfaces

Yes, initially I thought calling `lb4 model` should also populate the interface, but I agree that it should be removed and left up to another CLI command.

> The answer to your Question: should all models include a relations interface? is YES.
> These new `lb4 model` prompts should be removed.

👍 

> Ideally, `lb4 repository` should be improved to add entries to relations interfaces. This was not part of the original acceptance criteria and I don't know how much effort would be required, it may be best to leave this part out of scope of this pull request and possibly out of scope of #2960 too.

I originally thought it was, but I misunderstood the acceptance criteria, but I'm good with opening a new issue for this as it's out of scope of this PR.

Reply:
Issue created for adding entries to the interfaces: https://github.com/strongloop/loopback-next/issues/2988

In the new commits, I added empty interfaces because I think they should be populated only after https://github.com/strongloop/loopback-next/issues/2633 is landed. What do you think?